pkg/cmd/pulumi/config.go

```diff
@@ -326,6 +328,56 @@ func newConfigRmCmd(stack *string) *cobra.Command {
 	return rmCmd
 }
 
+func newConfigRmAllCmd(stack *string) *cobra.Command {
+	var path bool
+
+	rmAllCmd := &cobra.Command{
+		Use:   "rm-all <key1> <key2> <key3> ...",
+		Short: "Remove multiple configuration values",
+		Long: "Remove multiple configuration values.\n\n" +
+			"The `--path` flag indicates that keys should be parsed within maps or lists:\n\n" +
+			"  - `pulumi config rm-all outer.inner foo[0] key1` will remove the " +
+			"`outer.inner`, `foo[0]` and `key1` keys\n" +
+			"  - `pulumi config rm-all --path  outer.inner foo[0] key1` will remove the \n" +
+			"      `inner` key of the `outer` map, the first key of the `foo` list and `key1`.",
+		Args: cmdutil.MinimumNArgs(1),
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			opts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+
+			s, err := requireStack(*stack, true, opts, true /*setCurrent*/)
+			if err != nil {
+				return err
+			}
+
+			ps, err := loadProjectStack(s)
+			if err != nil {
+				return err
+			}
+
+			for _, arg := range args {
+				key, err := parseConfigKey(arg)
+				if err != nil {
+					return errors.Wrap(err, "invalid configuration key")
+				}
+
+				err = ps.Config.Remove(key, path)
+				if err != nil {
+					return err
+				}
+			}
+
+			return saveProjectStack(s, ps)
```

Comment:
There's a network call every time there's a secret encrypted or decrypted... but other than that it seems that the network call only happens when there's an `up`.

Reply:
I read through the code again, and I don't think this is going to help with the underlying latency issues. Secrets are the bottle-neck here it seems and this implementation loops through encrypting values one at a time for `set-all --secret`, which should be more or less equivalent to running `pulumi config set --secret` sequentially. We might need to think about spawing a go routine to do this in parallel, or creating a bulk config encryption endpoint in the service if one doesn't already exist.

It would be helpful to do some benchmarking here to understand performance impacts for a few different scenarios (bulk set plaintext, bulk set a bunch of secrets, etc).

In addition, the other underlying issue that I thought we were tackling with this, but didn't see mentioned in https://github.com/pulumi/pulumi/issues/6329 was the parallelism issues with setting config. Running config command async in a loop across stacks was causing things to get garbled because stack selection was changing. Does this change help with that? I notice that these two methods use `setCurrent true` when getting the stack so I'm not sure. Would probably be good to test as well.

Reply:
I know with certainty that this greatly improves the latency experienced by automation api: https://github.com/pulumi/pulumi/issues/6042#issuecomment-781807632
But yes, I'll consider the go routine as an additional improvement if this isn't satisfactory.

> I notice that these two methods use setCurrent true when getting the stack so I'm not sure.
That's a mistake, it should be `false`. In fact, `set-all` is false, I just made an error here.

Reply:
Okay, since my other profiling test was in python, I did some in Go this time.

Test: setting 20 config keys
Before:
* Mixed (10 secret, 10 plain): 11.69s
* Plain (20 plain): 11.25s
* Secret (20 secret): 12.29s

After:
* Mixed (10 secret, 10 plain): 1.18s
* Plain (20 plain): 1.05s
* Secret (20 secret): 1.51s

I think the results speak for themselves.

Reply:
Sorry I missed that profiling data in the linked issue. Glad to hear we've got this sorted out!